### PR TITLE
[13.0][IMP] product_cost_security: Add security to Purchase page on product sheet

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT29577

Please @sergio-teruel @victoralmau review this

### CODE REVIEW

File: product_cost_security/views/product_views.xml [hunk 1]
```diff
@@ -10,6 +10,11 @@
                     name="groups"
                 >product_cost_security.group_product_cost</attribute>
             </xpath>
+            <xpath expr="//page[@name='purchase']" position="attributes">
+                <attribute
+                    name="groups"
+                >product_cost_security.group_product_cost</attribute>
+            </xpath>
```
Comment (victoralmau) [inline product_cost_security/views/product_views.xml:17]:
```suggestion
            <xpath expr="//page[@name='purchase']" position="attributes">
                <attribute
                    name="groups"
                >purchase.group_purchase_manager</attribute>
            </xpath>
```

Although its correct, maybe it's a good idea create new addon to set groups to purchase page. What do you think?

Comment (CarlosRoca13) [reply product_cost_security/views/product_views.xml:17]: On the Purchase page we have the cost associated to each vendor, so I think that security group is correct... But I prefer the opinion of 1 more... What do you think @sergio-teruel ??
image

Comment (chienandalu) [reply product_cost_security/views/product_views.xml:17]: Or maybe just set the group to the field in the tree element

Comment (sergio-teruel) [reply product_cost_security/views/product_views.xml:17]: Normally, purchase users must have permission to view the cost of the product, so I think that this is not problem.
Maybe You can try add to group_purchase_user the security cost group as implied_ids, by this way, a purchase user always will have the security_cost group.

File: product_cost_security/README.rst [hunk 2]
```diff
@@ -47,7 +47,7 @@ Usage
 To use this module you need to:
 
 #. Go to product form view logged with this user and you will see the
-   standard_price field.
+   standard_price field and Purchase window.
```
Comment (sergio-teruel) [inline product_cost_security/README.rst]: and Purchase tab.

File: product_cost_security/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -1,2 +1,2 @@
-Adds a security group for product cost price field.
-Only users with this group can view this field.
+Adds a security group for product cost price field and purchase window on products
```
Comment (sergio-teruel) [inline product_cost_security/readme/DESCRIPTION.rst]: purchase tab

File: product_cost_security/readme/USAGE.rst [hunk 4]
```diff
@@ -1,4 +1,4 @@
 To use this module you need to:
 
 #. Go to product form view logged with this user and you will see the
-   standard_price field.
+   standard_price field and Purchase window.
```
Comment (sergio-teruel) [inline product_cost_security/readme/USAGE.rst]: the same

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
Okey so, changes done

File: product_cost_security/__manifest__.py [hunk 5]
```diff
@@ -12,6 +12,6 @@
     "license": "AGPL-3",
     "application": False,
     "installable": True,
-    "depends": ["product"],
+    "depends": ["product", "purchase"],
```
Comment (pedrobaeza) [inline product_cost_security/__manifest__.py]: I don't see correct that this module depends on purchase.

**Comment by CarlosRoca13** [comment]
What do you suggest @pedrobaeza ?

File: product_cost_security/__manifest__.py (hunk 5 above)
Comment (pedrobaeza) [reply product_cost_security/__manifest__.py]: A new glue module should be done.

**Pull request closed without merging**